server/src/main/java/org/apache/druid/segment/loading/SegmentLocalCacheManager.java

```diff
@@ -225,22 +227,50 @@ public List<DataSegment> getCachedSegments() throws IOException
       }
     }
 
-    final List<DataSegment> cachedSegments = new ArrayList<>();
+    final ConcurrentLinkedQueue<DataSegment> cachedSegments = new ConcurrentLinkedQueue<>();
     final File[] segmentsToLoad = retrieveSegmentMetadataFiles();
+    final CountDownLatch latch = new CountDownLatch(segmentsToLoad.length);
+
+    // If there is no dedicated bootstrap executor, perform the loading sequentially on the current thread.
+    final boolean isLoadingSegmentCacheFromDiskWithoutBootstrapExecutor = loadOnBootstrapExec == null;
```

Comment:
Nit: seems a little verbose 🙂 , especially since it is just to be used in this method.

You may call this variable `loadSegmentsSequentially` or `useBootstrapLoadExec` (with the inverse condition).

Reply:
Sounds good, I renamed it to isLoadingSegmentsSequentially. Thanks!